arkouda/client.py

```diff
@@ -327,21 +326,19 @@ def _send_string_message(cmd : str, recv_bytes : bool=False,
         expected fields       
     """
     message = RequestMessage(user=username, token=token, cmd=cmd, 
-                          format=MessageFormat.STRING, args=cast(str,args))
+                          format=MessageFormat.STRING, args=args)
 
     logger.debug('sending message {}'.format(message))
 
     socket.send_string(json.dumps(message.asdict()))
 
-    if recv_bytes:
-        return_message = socket.recv()
-
-        # raise errors or warnings sent back from the server
-        if return_message.startswith(b"Error:"): 
-            raise RuntimeError(return_message.decode())
-        elif return_message.startswith(b"Warning:"): 
-            warnings.warn(return_message.decode())
-        return return_message
+    if recv_binary:
+        frame = socket.recv(copy=False)
+        view = frame.buffer
+        # raise errors sent back from the server
+        if bytes(view[0:len(b"Error:")]) == b"Error:":
+            raise RuntimeError(frame.bytes.decode())
+        return view
```

Comment:
I noticed handling of "Warning" was dropped here (as well as later on ~line 400-409).  I haven't really looked into it but were we never actually returning a Warning string so the handling was never reached?

Reply:
Noted in https://github.com/Bears-R-Us/arkouda/pull/885/commits/32eba00ba480648ff251e4a48840a403e6f32aa1, but this code never really supported warnings. We could warn here, but then would later fail to convert the binary message to numpy array because the raw buffer would still have the warning. We'd need some way to separate the warning from the binary message and for now I decided to just remove the warning capability since it isn't used.